book/src/rfcs/benchmarking.md

```diff
@@ -0,0 +1,266 @@
+| kimchi memory flamegraph | proof-systems | memory   | no               | yes              | no              | no               |
+| snarky time              | mina/snarky*  | time     | yes              | yes              | yes             | yes              |
+| snarky memory            | mina/snarky*  | memory   | yes              | yes              | yes             | no               |
+| pickles time             | mina/pickles* | time     | yes              | yes              | yes             | yes              |
+| pickles memory           | mina/pickles* | memory   | yes              | yes              | yes             | no               |
+
+\* these have to be in Mina now because pickles is in Mina, and while snarky has
+its own repository, using it requires a few things still in Mina, so that we
+might want to consider some changes to make these benchmarks easier and probably
+cheaper.
+
+### Kimchi time
+
+One of the simplest measuring running time, can be a specific case or allow more
+flexibility by introducing parameters like for example different circuit sizes.
+We already have most of the infrastructure for this with
+[criterion](https://crates.io/crates/criterion). We would ideally have a
+dedicated computer. However, considering that there aren't that many PRs in
+proof-systems, the requirements shouldn't be that high.
+
+### Kimchi time flamegraph
+
+This can be considered a variant of the [previous](#kimchi-time). Some more
+research is required but this information ideally can be generated from the
+already existing benchmarks with little to no changes.  This may not be as
+interesting to have in CI, but can be useful to run locally for those working in
+optimizations.
+
+### Kimchi instructions
+
+This is basically counting the CPU instructions that takes to run some code,
+could be use as an alternative for CI given that it won't be affected by noise
+and can work in any computer. It would also be faster by running the code only
+once. The implementation will require a bit more research but one option is the
+crate [iai](https://crates.io/crates/iai), with some thinking it could be
+possible to have this and the time benchmarks share the same code.  Is worth
+mentioning that the result are specific to the code and even the computer
+running the benchmark and thus not useful for comparisons with other projects.
+
+### Kimchi memory
+
+Similar to time, but with memory, the constraints are different in some ways,
+noise is not generally a problem and any computer can be used, and also requires
+less sampling and should run faster. This is to some degree already implemented
+through a custom measurement for criterion, and in practice can be slower than
+measuring time, the instrumentation makes it run slower and the some limitations
+in criterion require to run the code more than necessary. With that in mind, the
```

Comment:
Which are the subset? Can you be specific?

Reply:
I don't think it can be specified at this point, it would depend on how much each benchmark costs, (both time and money), cost of all benchmarks, how much do we want to spend and what benchmarks seem more valuable relative to others.
We could start with running every single benchmark and removing the least valuables until we are under budget.